app/components/filters_component.rb

```diff
@@ -1,34 +1,57 @@
 class FiltersComponent < GovukComponent::Base
-  attr_accessor :filters, :form, :items, :options
+  attr_accessor :filters, :form, :options
 
   def self.variants
     %w[default]
   end
 
-  def initialize(filters:, form:, items:, options:, classes: [], html_attributes: {})
+  def initialize(form:, options:, filters: {}, classes: [], html_attributes: {})
     super(classes: classes, html_attributes: html_attributes)
 
     @filters = filters
     @form = form
-    @items = items
     @options = options
   end
 
-  def render?
-    filters.present?
-  end
+  renders_many :groups, lambda { |key:, component:|
+    tag.div(key, class: "filters-component__groups__group", data: { group: key }) do
+      tag.div(component)
+    end
+  }
 
-  def display_remove_buttons?
-    filters[:total_count].positive? && options[:remove_buttons]
+  renders_one :remove_buttons, "RemoveButtons"
+
+  class RemoveButtons < GovukComponent::Base
+    def initialize(classes: [], html_attributes: {})
```

Comment:
If you like, this can inherit from `ApplicationComponent` which will give you this same initializer.